mkosi

```diff
@@ -2313,14 +2313,27 @@ def detect_distribution():
 
     id = None
     version_id = None
+    codename = None
 
     for ln in f:
         if ln.startswith("ID="):
             id = ln[3:].strip()
         if ln.startswith("VERSION_ID="):
             version_id = ln[11:].strip()
+        if ln.startswith("VERSION="):
+            # extract Debain release codename
+            version_str = ln[8:].strip()
+            open_paren_pos = version_str.index("(")
+            close_paren_pos = version_str.index(")")
+
+            if open_paren_pos and close_paren_pos:
+                codename = version_str[(open_paren_pos+1):close_paren_pos]
```

Comment:
hmm, doesn't debian set VERSION_CODENAME= for this? this was added explicitly for this purpose... if they don't set this right now, I think we should still look for it and use preferably it if it's set, and also somebody should file a bug requesting them to add this for cases like this...

Reply:
see docs about this:

https://www.freedesktop.org/software/systemd/man/os-release.html#VERSION_CODENAME=